cli/ecoli.c

```diff
@@ -162,3 +162,44 @@ int arg_eth_addr(const struct ec_pnode *p, const char *id, struct rte_ether_addr
 
 	return 0;
 }
+
+static int arg_ip(const struct ec_pnode *p, const char *id, void *addr, int family) {
+	const char *str = arg_str(p, id);
+	if (str == NULL)
+		return -errno;
+
+	if (inet_pton(family, str, addr) != 1)
+		return errno_set(EINVAL);
+
+	return 0;
+}
+
+int arg_ip4(const struct ec_pnode *p, const char *id, ip4_addr_t *addr) {
+	return arg_ip(p, id, addr, AF_INET);
+}
+
+int arg_ip6(const struct ec_pnode *p, const char *id, struct rte_ipv6_addr *addr) {
+	return arg_ip(p, id, addr, AF_INET6);
+}
+
+static int
+arg_ip_net(const struct ec_pnode *p, const char *id, void *net, bool zero_mask, int family) {
+	const char *str = arg_str(p, id);
+	if (str == NULL)
+		return -errno;
+
+	if (family == AF_INET)
+		return ip4_net_parse(str, net, zero_mask);
+	else
+		return ip6_net_parse(str, net, zero_mask);
+
+	return 0;
```

Comment:
nit pick, `return 0` is unreachable, can you do this instead?

```c
switch (family)
case AF_INET:
        return ip4_net_parse(str, net, zero_mask);
case AF_INET6:
        return ip6_net_parse(str, net, zero_mask);
default:
        return errno_set(EINVAL);
}
```

Reply:
I know this function is static and is not called with any other family value. But this cannot hurt and helps with readablility.